repo2docker/app.py

```diff
@@ -650,6 +651,19 @@ def build(self):
                                extra=dict(phase='building'))
 
                 if not self.dry_run:
+                    if self.user_id == 0:
+                        self.log.error(
+                            'Root as the primary user in the image is not permitted.\n'
+                        )
+                        self.log.info(
+                            "The uid and the username of the user invoking repo2docker "
+                            "is used to create a mirror account in the image by default. "
+                            "To override that behavior pass --user-id <numeric_id> and "
+                            " --user-name <string> to repo2docker.\n"
+                            "Please see repo2docker --help for more details.\n"
+                        )
+                        sys.exit(errno.EPERM)
```

Comment:
Since r2d can be used as a library, this code should raise an exception that's caught elsewhere and the message printed. Otherwise, the calling code just exits and can not actually handle anything here.

See https://github.com/jupyter/repo2docker/issues/696.

Previously, #496